yt_dlp/extractor/nzonscreen.py

```diff
@@ -45,49 +49,109 @@ class NZOnScreenIE(InfoExtractor):
+    }, {
+        'url': 'https://www.nzonscreen.com/title/reluctant-hero-2008',
+        'info_dict': {
+            'id': '847f5c91af65d44b',
+            'ext': 'mp4',
+            'format_id': 'hi',
+            'height': 360,
+            'width': 640,
+            'subtitles': {
+                'en': [{'ext': 'vtt', 'url': 'https://www.nzonscreen.com/captions/3367'}],
+            },
+            'title': 'Reluctant Hero (clip 1)',
+            'description': 'Part one of four from this full length documentary.',
+            'display_id': 'reluctant-hero-2008',
+            'duration': 1108.0,
+            'alt_title': 'Reluctant Hero',
+            'thumbnail': r're:https://www\.nzonscreen\.com/content/images/.+\.jpg',
+        },
+        'params': {'noplaylist': True},
     }]
 
     def _extract_formats(self, playlist):
+        formats = []
         for quality, (id_, url) in enumerate(traverse_obj(
-                playlist, ('h264', {'lo': 'lo_res', 'hi': 'hi_res'}), expected_type=url_or_none).items()):
-            yield {
-                'url': url,
-                'format_id': id_,
-                'ext': 'mp4',
-                'quality': quality,
-                'height': int_or_none(playlist.get('height')) if id_ == 'hi' else None,
-                'width': int_or_none(playlist.get('width')) if id_ == 'hi' else None,
-                'filesize_approx': float_or_none(traverse_obj(playlist, ('h264', f'{id_}_res_mb')), invscale=1024**2),
-            }
+                playlist, ('h264', {'lo': 'lo_res', 'hi': 'hi_res', 'hd': 'hd_res'}),
+                expected_type=url_or_none).items()):
+            if traverse_obj(playlist, ('h264', f'{id_}_res_mb', {float_or_none})):
+                formats.append({
+                    'url': url,
+                    'format_id': id_,
+                    'ext': 'mp4',
+                    'quality': quality,
+                    'filesize_approx': float_or_none(traverse_obj(
+                        playlist, ('h264', f'{id_}_res_mb')), invscale=1024**2),
+                })
+        if formats:
+            formats[-1].update(traverse_obj(playlist, {
+                'height': ('height', {int_or_none}),
+                'width': ('width', {int_or_none}),
+            }))
+        return formats
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
+        title = strip_or_none((
+            self._html_extract_title(webpage, default=None)
+            or self._og_search_title(webpage)).rsplit('|', 2)[0])
+        playlist = self._download_json(
+            f'https://www.nzonscreen.com/html5/video_data/{video_id}', video_id,
+            'Downloading media data')
 
-        playlist = self._parse_json(self._html_search_regex(
-            r'data-video-config=\'([^\']+)\'', webpage, 'media data'), video_id)
+        if not self._yes_playlist(video_id, video_id):
+            del playlist[1:]
```

Comment:
With `--no-playlist`, the IE should return only the video result, not `playlist_result`